### PR TITLE
Move the extending_theano documentation out of tutorial

### LINKED ISSUE #3670: Move the creating ops part of tutorial to the extending section.

### PR DESCRIPTION
**Pull request by fvisin**
Fixes #3670 

@abergeron I tried to split the documentation in sections to make it more readable, let me know if they seem reasonable.

### CODE REVIEW

File: doc/extending/extending_theano.txt [hunk 1]
```diff
@@ -422,7 +505,8 @@ and ``b`` are equal.
 
         def make_node(self, x):
             # check that the theano version has support for __props__.
-            assert hasattr(self, '_props'), "Your version of theano is too old to support __props__."
+            assert hasattr(self, '_props'), "Your version of theano is too old 
+            to support __props__."
```
Comment (abergeron) [inline doc/extending/extending_theano.txt]: I know this was in the original code, but I would remove it.  Theano has had support for __props__ for long enough.

**Comment by abergeron** [comment]
I like the new organization.  I feel it's much better.

I'll let @nouiz and @lamblin comment, but otherwise, I'm ok with merging this.

File: doc/extending/extending_theano.txt [hunk 2]
```diff
@@ -1,24 +1,35 @@
 
-.. _extending_theano:
```
Comment (lamblin) [inline doc/extending/extending_theano.txt]: Should we keep that label so that the file can more easily be referenced from the table of content?

**Comment by lamblin** [comment]
Otherwise, that's good for me.

**Comment by fvisin** [comment]
I am glad you like the new structure!
I am incorporating your suggestions and checking for broken links. Don't merge it yet, I'll commit a few minor fixes soon.

**Comment by fvisin** [comment]
I committed a few fixes. 
I am not able to overcome this error, I don't get what's the problem. There are 3 `.. code-block::` directives and they all seem ok to me. 

``` bash
python doc/scripts/docgen.py --nopdf --cache
Running Sphinx v1.3.1
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 173 source files that are out of date
updating environment: 173 added, 0 changed, 0 removed
/data/lisa/exp/visin/theano/theano/theano/tensor/signal/downsample.py:5: UserWarning: downsample module has been moved to the pool module.                                                                                                    
  warnings.warn("downsample module has been moved to the pool module.")
reading sources... [100%] tutorial/using_multi_gpu                                                                                                                                                                                            

Warning, treated as error:
/data/lisa/exp/visin/theano/theano/doc/extending/extending_theano_c.txt:1035: ERROR: Error in "code-block" directive:
1 argument(s) required, 0 supplied.

.. code-block::

    $gdb python
    (gdb)r bin/theano-nose theano/
```

**Comment by abergeron** [comment]
You have to set a code type for the code block. In this case it would be sh as in 

```
.. code-block:: sh
```

**Comment by fvisin** [comment]
Oh gosh....I was looking into `extending_theano.txt` instead of `extending_theano_c.txt`, no wonder I didn't find the error. Thanks

**Comment by abergeron** [comment]
Ok for me when travis passes.

**Comment by fvisin** [comment]
I noticed there are 2 graph related section so I am merging them. I'll commit soon.

**Comment by fvisin** [comment]
Ok, I finally fixed all the errors (many of which did not depend on my modifications). Now the docs compile properly, at least locally. Tell me if you want me to collapse some of the commits together, otherwise it can be merged.

**Comment by lamblin** [comment]
This PR needs a rebase.

**Comment by fvisin** [comment]
Oh...damn it! 
Ok. I don't have time right now, but I will incorporate your comments and rebase soon.

**Comment by fvisin** [comment]
I have fixed everything and rebased. It should be ready for merge.

**Comment by nouiz** [comment]
As this remove some doc file, old url/bookmark are broken.

Just making a new file with the old name that tell in that page, moved and a link to the new place would fix this problem.

**Comment by fvisin** [comment]
Before pushing the PR I generated the documentation locally to check that there were no broken links, but I didn't think about external bookmarks/urls. I will do another PR with the links you suggested soon.

**Pull request merged**